src/Microsoft.DotNet.Cli.Utils/CommandResolution/DefaultCommandResolver.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class DefaultCommandResolver : ICommandResolver
+    {
+        private IList<ICommandResolver> _orderedCommandResolvers;
+
+        public DefaultCommandResolver() : this(new PackagedCommandSpecCreator()) { }
+
+        public DefaultCommandResolver(IPackagedCommandSpecCreator packagedCommandSpecCreator)
```

Comment:
I'm not sure I follow. Move the constructor to an extension method or move the default set of injected classes to an extension method?